cmd/utils.go

```diff
@@ -145,3 +111,44 @@ func sliceContains(val []Chain, key Chain) bool {
 	}
 	return false
 }
+
+func generateGraph(goModGraphOutputString string ) DependencyOverview {
+	depGraph := DependencyOverview{MainModules: mainModules}
```

Comment:
`mainModules` was being passed as an argument to the `getDepInfo` function so we won't have access to that here.

Reply:
So can I keep the signature like 
generateGraph(goModGraphOutputString string, mainModules []string) DependencyOverview 
any suggestions ?

Reply:
No need for that. You can add the main modules after line 55. Something like:

```
depGraph := generateGraph(goModGraphOutputString)
depGraph.MainModules = mainModules
return &depGraph
```

Reply:
But If we do what you are suggesting then the graph would be generated according to the go mod graph output which assumes the first module to be the mainModule and later we simply overwrite the mainModule as opposed to generating the graph for arbitrary mainModule .
For example:
depstat stats -m "C"
go mod graph output:
A B
A C
C D
ourOutput(If we do not inject the mainModule in generateGraph logic) 
mainModule: C
directDependency: B C

O/P in case we inject mainModule in generateGraph func:
mainModule: C
directDependency: D

Please let me know your thoughts ?

Reply:
Ah yes you're right I forgot that we were using `mainModules` in the `generateGraph` function. Best change the function signature then :)